src/main/java/games/strategy/engine/data/properties/GameProperties.java

```diff
@@ -154,41 +152,54 @@ public IEditableProperty getPlayerProperty(final String name) {
     return playerProperties.get(name);
   }
 
-  public static void toOutputStream(final OutputStream sink, final List<IEditableProperty> editableProperties)
-      throws IOException {
-    // write internally first in case of error
-    try (ByteArrayOutputStream bos = new ByteArrayOutputStream(5000);
-        ObjectOutputStream outStream = new ObjectOutputStream(bos);
-        GZIPOutputStream zippedOut = new GZIPOutputStream(sink)) {
-
-      outStream.writeObject(editableProperties);
-      zippedOut.write(bos.toByteArray());
-      zippedOut.flush();
-    }
```

Comment:
There was a potential bug here in that `outStream` was not being closed before `bos.toByteArray()` was called.